pilot/pkg/bootstrap/namespacecontroller.go

```diff
@@ -116,6 +125,31 @@ func NewNamespaceController(data func() map[string]string, kubeClient kubernetes
 				return c.namespaceChange(obj)
 			})
 		},
+		UpdateFunc: func(newObj, oldObj interface{}) {
+			c.queue.Push(func() error {
+				return c.namespaceChange(newObj)
+			})
+		},
+		DeleteFunc: func(obj interface{}) {
+			ns, ok := obj.(*v1.Namespace)
+			if !ok {
+				tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+				if !ok {
+					log.Errorf("error decoding object, invalid type")
+					return
+				}
+
+				ns, ok = tombstone.Obj.(*v1.Namespace)
+				if !ok {
+					log.Errorf("error decoding object tombstone, invalid type")
+					return
+				}
+			}
+
+			c.queue.Push(func() error {
+				return c.namespaceChange(ns)
```

Comment:
similar question about update, we probably don't want to write to a deleted namespace as it will fail to write since its terminating?